spacy/tests/serialize/test_serialize_kb.py

```diff
@@ -112,6 +114,47 @@ def __init__(self, vocab, entity_vector_length, custom_field):
             super().__init__(vocab, entity_vector_length)
             self.custom_field = custom_field
 
+        def to_disk(self, path, exclude: Iterable[str] = SimpleFrozenList()):
+            """We overwrite InMemoryLookupKB.to_disk() to ensure that self.custom_field is stored as well."""
+            path = ensure_path(path)
+            if not path.exists():
+                path.mkdir(parents=True)
+            if not path.is_dir():
+                raise ValueError(Errors.E928.format(loc=path))
+
+            def serialize_custom_fields(
+                values: Dict[str, Any], file_path: Path
+            ) -> None:
+                with open(file_path, "wb") as file:
+                    pickle.dump(values, file)
```

Comment:
We could do something similar to what the pipeline components do - they have an internal `self.cfg` dictionarity that needs to have json-serializable values, and then we can use `srsly.json_loads` and `srsly.write_json`

Reply:
`self.cfg` would be `{"custom_field": self.custom_field}` then? Or do we want to drop `self.custom_field` and only keep `self.cfg` (with entry `custom_field`)?

Reply:
In this particular case we don't really need to define a `self.cfg`, we can just compile the dict on the fly like you did. I wanted to reference this so you could have a look how it's done in the components and how they use `srsly` .

Reply:
Avoiding `pickle` is sound advice - I tripped over that recently with some `wikid` stuff :upside_down_face: 
Updated in https://github.com/explosion/spaCy/pull/12320/commits/3a58c19cc1419f1b15dd213461750061da846539.